### PR TITLE
Fix styles that should not be made dynamic

### LINKED ISSUE #30799: styled-jsx issue with interpolation when SWC is enabled

### PR DESCRIPTION
**Pull request by padmaia**
Fixes #30799

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: packages/next/build/swc/src/styled_jsx/mod.rs [hunk 1]
```diff
@@ -495,7 +504,16 @@ impl StyledJSXTransformer {
...
     }
 
-    fn process_tagged_template_expr(&mut self, tagged_tpl: &TaggedTpl) -> Result<Expr, Error> {
+    fn process_tagged_template_expr(
+        &mut self,
+        tagged_tpl: &TaggedTpl,
+        tag: &str,
+    ) -> Result<Expr, Error> {
+        if tag != "resolve" {
+            // Check whether there are undefined references or
```
Comment (padmaia) [inline packages/next/build/swc/src/styled_jsx/mod.rs:513]: I will work on this in a follow up

File: packages/next/build/swc/tests/fixture/styled-jsx/styles/input.js [hunk 2]
```diff
@@ -24,7 +24,6 @@ export const uh = bar
 
 export const foo = css`div { color: ${color}}`
 
-// TODO: the next 3 should not transformed as dynamic
```
Comment (padmaia) [inline packages/next/build/swc/tests/fixture/styled-jsx/styles/input.js:27]: It was actually only the next 2 that didn't need to be dynamic.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| buildDuration | 22.9s | 22.3s | -593ms |
| buildDurationCached | 4.5s | 4.5s | ⚠️ +27ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +1 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.632 | 3.793 | ⚠️ +0.16 |
| / avg req/sec | 688.39 | 659.11 | ⚠️ -29.28 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.942 | 2.024 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1287.39 | 1234.95 | ⚠️ -52.44 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| index.html gzip | 520 B | 520 B | ✓ |
| link.html gzip | 531 B | 531 B | ✓ |
| withRouter.html gzip | 513 B | 513 B | ✓ |
| Overall change | 1.56 kB | 1.56 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| buildDuration | 24.7s | 23.4s | -1.3s |
| buildDurationCached | 4.4s | 4.4s | -16ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +1 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.665 | 3.637 | -0.03 |
| / avg req/sec | 682.22 | 687.46 | +5.24 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.956 | 1.899 | -0.06 |
| /error-in-render avg req/sec | 1278.31 | 1316.37 | +38.06 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js fix/styled-jsx-dynamic-external | Change |
| - | - | - | - |
| index.html gzip | 523 B | 523 B | ✓ |
| link.html gzip | 536 B | 536 B | ✓ |
| withRouter.html gzip | 517 B | 517 B | ✓ |
| Overall change | 1.58 kB | 1.58 kB | ✓ |

</details>

</details>
Commit: 036604d7ab8b3d8fd13026a24a45c83a3cbe6513

**Review by kdy1 — APPROVED** [review]
Nice work!

**Pull request merged**